### DISCUSSION TITLE
Soup salt exception

### DISCUSSION THREAD

**Discussion by jrlane** [Security-Onion-Solutions/securityonion#6431] [Unsupported Versions] [answered]
Running soup on the manager generates the error posted below. Running soup multiple times does not change the result. I have also attached a full output. Any ideas?

```
This is a manager, so we can proceed.
Found that Security Onion 2.3.70 is currently installed.

Cloning Security Onion github repo into /tmp/sogh/securityonion.
Removing previous upgrade sources.
Cloning the Security Onion Repo.
Cloning into 'securityonion'...
remote: Enumerating objects: 49771, done.
remote: Counting objects: 100% (5140/5140), done.
remote: Compressing objects: 100% (1523/1523), done.
remote: Total 49771 (delta 3453), reused 4981 (delta 3336), pack-reused 44631
Receiving objects: 100% (49771/49771), 15.05 MiB | 14.90 MiB/s, done.
Resolving deltas: 100% (31867/31867), done.
Verifying we have the latest soup script.
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
UnicodeEncodeError: 'ascii' codec can't encode character '\u2019' in position 37179: ordinal not in range(128)
Traceback (most recent call last):
  File "/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3003', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 129, in run
    _retcode=ret.get("retcode", 0),
  File "/usr/lib/python3.6/site-packages/salt/output/__init__.py", line 122, in display_output
    ofh.write(salt.utils.stringutils.to_str(fdata))
UnicodeEncodeError: 'ascii' codec can't encode character '\u2019' in position 37179: ordinal not in range(128)
```
[soup-out.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7637703/soup-out.txt)

**Comment by tiger-greenhell** [comment, marked as answer]
> Running soup on the manager generates the error posted below. Running soup multiple times does not change the result. I have also attached a full output. Any ideas?
> 
> ```
> This is a manager, so we can proceed.
> Found that Security Onion 2.3.70 is currently installed.
> 
> Cloning Security Onion github repo into /tmp/sogh/securityonion.
> Removing previous upgrade sources.
> Cloning the Security Onion Repo.
> Cloning into 'securityonion'...
> remote: Enumerating objects: 49771, done.
> remote: Counting objects: 100% (5140/5140), done.
> remote: Compressing objects: 100% (1523/1523), done.
> remote: Total 49771 (delta 3453), reused 4981 (delta 3336), pack-reused 44631
> Receiving objects: 100% (49771/49771), 15.05 MiB | 14.90 MiB/s, done.
> Resolving deltas: 100% (31867/31867), done.
> Verifying we have the latest soup script.
> [ERROR   ] An un-handled exception was caught by salt's global exception handler:
> UnicodeEncodeError: 'ascii' codec can't encode character '\u2019' in position 37179: ordinal not in range(128)
> Traceback (most recent call last):
>   File "/bin/salt-call", line 11, in <module>
>     load_entry_point('salt==3003', 'console_scripts', 'salt-call')()
>   File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
>     client.run()
>   File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
>     caller.run()
>   File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 129, in run
>     _retcode=ret.get("retcode", 0),
>   File "/usr/lib/python3.6/site-packages/salt/output/__init__.py", line 122, in display_output
>     ofh.write(salt.utils.stringutils.to_str(fdata))
> UnicodeEncodeError: 'ascii' codec can't encode character '\u2019' in position 37179: ordinal not in range(128)
> ```
> 
> [soup-out.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7637703/soup-out.txt)

I confirm the same problem, Security Onion 2.3.80, Centos7

The problem start up after upgrading to the latest version of soup.

The problem did not appear on Ubuntu 18.04 + Security Onion 2.3.80

[soup.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7641254/soup.txt)

**Reply by gitmffonseca** [reply]
For me the solution was the following command at console:

_export LC_CTYPE="en_US.UTF-8"_

**Reply by jrlane** [reply]
Thanks for the reply that does indeed allow the script to run successfully! I do however see this line in the beginning as a result of the change.

"Failed to set locale, defaulting to C"

**Reply by gitmffonseca** [reply]
Hi @jrlane !

Maybe for that specific warning you cloud execute the following line

_sudo localectl set-locale LANG=en_US.UTF-8_

**Comment by m0duspwnens** [comment]
What was the locale set to prior to changing it?

Thanks.

**Reply by tiger-greenhell** [reply]
C.UTF-8

**Reply by m0duspwnens** [reply]
Thank you. We are working on a hotfix that will likely be released next week. This is one of the issues that will be addressed in it.

**Comment by kmfreder1** [comment]
I was having the same error as above on my Ubuntu 18.04 install.  

I also saw this error in the console at the very top of the Soup run:

`/usr/sbin/soup: line 19: warning: setlocale: LC_CTYPE: cannot change locale (en_US.UTF-8): No such file or directory`

After coming here, I tried to run:

 `export LC_CTYPE="en_US.UTF-8"` 

but I got the following error:

`-su: warning: setlocale: LC_CTYPE: cannot change locale (en_US.UTF-8): No such file or directory`

So, I searched that error and found [this](https://askubuntu.com/questions/454260/how-to-solve-locale-problem)

From that link, I ran the following:

`sudo locale-gen en_US.UTF-8`

And then soup ran without the above errors.  

Interestingly, I performed an upgrade on another host that was also 2.3.80 on Ubuntu 18 and I didn't get either of those errors.  The only difference between those two hosts is that the one had originally been installed with 2.3.50 or 2.3.60 and upgraded from there over time and the other started out on 2.3.70.  

I don't know if that provides any insight for anyone, but for what it's worth, that's what worked for me.  

Thanks SO Team for responding so quickly to the log4j stuff!

**Reply by dougburks** [reply]
@m0duspwnens created issue https://github.com/Security-Onion-Solutions/securityonion/issues/6599 for this.

**Reply by kmfreder1** [reply]
I saw that!  Thanks for the note.  Really appreciate the work you guys do and your responsiveness.
